pkg/core/database/heavy/transactions.go

```diff
@@ -155,11 +155,13 @@ func (t transaction) StoreBlock(b *block.Block) error {
 		// Schema
 		//
 		// Key = OutputKeyPrefix + tx.output.PublicKey
-		// Value = tx.output.PublicKey
+		// Value = unlockheight
 		//
 		// To make FetchOutputKey functioning
 		for _, output := range tx.StandardTx().Outputs {
-			t.put(append(OutputKeyPrefix, output.PubKey.P.Bytes()...), output.PubKey.P.Bytes())
+			value := make([]byte, 8)
```

Comment:
@julesdesmit Would you clarify the scenario from below?  What would happen if ...

0. Alice has 100 Dusk (in a single output) 
1. Alice sends a BidTx with amount of 10 Dusk, unlockHeight 55
2. As a result a tx with one input and two outputs is created
	- output_1 - 10 dusk (bid amount)
	- output_2 - 90 dusk (the change)
3. Both outputs will be stored into db key-value entries as
	output_1.PubKey, unlockHeight: 55
	output_2.PubKey, unlockHeight: 55

If the above is correct understanding, then Alice will not be capable of spending the `change` of 90 dusk until 55 height. So in that case even the bid amount is only 10 the entire bucket of 100 DUSK is locked.

Lemme know if I misunderstood it.

Reply:
That's a really good catch. This is currently a bit tricky to work around, since the unlock height is specified on the transaction, and not on a per-output basis. Maybe a slight refactoring of the tx structure is in order.

Reply:
I think we could work around it by only locking the first output of any tx, as we currently do not support multi-output txs.